src/installer/constants.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2025 Lunatic Fringers
+#
+# This file is part of Shepherd Core Stack
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+REQUIRED_PKGS = [
+    "bc",
+    "jq",
+    "curl",
+    "rsync",
+    "apt-transport-https",
+    "ca-certificates",
+    "software-properties-common",
+    "gnupg",
+    "lsb-release",
+]
+
+REQUIRED_PYTHON_PKGS = ["python3-venv", "python3-pip"]
+
+REQUIRED_DOCKER_PKGS = [
+    "docker.io",
+    "docker-ce",
+    "docker-ce-cli",
+    "containerd.io",
+    "docker-compose",
+    "docker-compose-plugin",
+]
+
+INSTALL_COMMANDS = {
+    "debian": ["sudo", "apt-get", "install", "-y"],
+    "ubuntu": ["sudo", "apt-get", "install", "-y"],
+    "fedora": ["sudo", "dnf", "install", "-y"],
+    "centos": ["sudo", "yum", "install", "-y"],
+    "arch": ["sudo", "pacman", "-S", "--noconfirm"],
+}
+
+UPDATE_COMMANDS = {
+    "debian": "sudo apt update",
```

Comment:
Do we really need to support all these distro in the first implementation?
How are you planning to test the installation over all these distro?
I suggest to start small, and support only what we can really test.